merkle/log_verifier_test.go

```diff
@@ -416,10 +416,102 @@ func TestVerifyConsistencyProof(t *testing.T) {
 	}
 }
 
+func getLeafAndProof(tree *InMemoryMerkleTree, index int64) ([]byte, [][]byte) {
+	// Note: InMemoryMerkleTree counts leaves from 1.
+	desc := tree.PathToCurrentRoot(index + 1)
+	proof := make([][]byte, len(desc))
+	for i, d := range desc {
+		proof[i] = d.Value.Hash()
+	}
+	leafHash := tree.LeafHash(index + 1)
+	return leafHash, proof
+}
+
+func TestPrefixHashFromInclusionProofGenerated(t *testing.T) {
+	sizes := make([]int64, 0, 18)
```

Comment:
I'd be tempted to run this up against a much larger number - we've all been bitten by Merkle tree bugs which only appeared at unexpectedly large tree sizes :)

Reply:
Done.